git_machete/cmd.py

```diff
@@ -266,47 +274,55 @@ def shell_escape(arg):
     return " ".join(env_repr + [cmd] + list(map(shell_escape, args)))
 
 
-def run_git(git_cmd, *args, **kwargs):
+def run_git(git_cmd: str, *args: str, **kwargs: Any) -> int:
     exit_code = run_cmd("git", git_cmd, *args, **kwargs)
     if not kwargs.get("allow_non_zero") and exit_code != 0:
         raise MacheteException(f"`{cmd_shell_repr('git', git_cmd, *args, **kwargs)}` returned {exit_code}")
     return exit_code
 
 
-def popen_git(git_cmd, *args, **kwargs):
+def popen_git(git_cmd: str, *args: str, **kwargs: Any) -> str:
```

Comment:
I'd enforce `str` for each of the kwarg values if possible

Reply:
Here and in other similar functions

Reply:
Cannot be be more precise in case of these two :c
- `def popen_cmd(cmd: str, *args: str, **kwargs: Any) -> Tuple[int, str, str]:`
- `def run_cmd(cmd: str, *args: str, **kwargs: Any) -> int:`

For the 3 others `Dict[str, str]` has been used